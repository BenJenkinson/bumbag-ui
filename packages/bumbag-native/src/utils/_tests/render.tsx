import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
