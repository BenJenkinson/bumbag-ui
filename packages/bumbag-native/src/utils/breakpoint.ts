import { Dimensions } from 'react-native';
import { theme } from './theme';

export function breakpoint(direction, values) {
  return (props) => {
    const breakpoints = theme('breakpoints')(props);
    const directionBreakpoints = breakpoints[direction];
    const size = Dimensions.get('window')[direction];

    const directionBreakpointsArray = Object.entries(directionBreakpoints).reverse();

    let hasMatched;
    let value = values.default;
    directionBreakpointsArray.forEach(([breakpoint, upperLimit], index) => {
      if (hasMatched) return;

      const lowerLimit = directionBreakpointsArray?.[index + 1]?.[1] || 0;

      // 'md', 'lg', etc takes first precedence
      if (size <= upperLimit && size > lowerLimit) {
        const matchedValue = values[breakpoint];
        value = matchedValue || value;
        hasMatched = Boolean(matchedValue);
      }

      // 'min-md', 'min-lg', etc takes second precedence
      if (size > lowerLimit) {
        const matcher = `min-${breakpoint}`;
        const matchedValue = values[matcher];
        value = matchedValue || value;
        hasMatched = Boolean(matchedValue);
      }

      // 'max-md', 'max-lg', etc takes third precedence
      if (size <= upperLimit) {
        const matcher = `max-${breakpoint}`;
        value = values[matcher] || value;
      }
    });

    return value;
  };
}
