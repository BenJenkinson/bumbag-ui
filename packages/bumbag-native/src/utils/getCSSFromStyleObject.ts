import _kebabCase from 'lodash/kebabCase';

import { css } from '../styled';
import {
  altitude,
  border,
  borderRadius,
  font,
  fontSize,
  lineHeight,
  palette,
  space,
  fontWeight,
  letterSpacing,
} from './theme';
import { Platform } from 'react-native';

const FLEX_HORIZONTAL_ALIGN_MAP = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
};

const FLEX_VERTICAL_ALIGN_MAP = {
  top: 'flex-start',
  center: 'center',
  bottom: 'flex-end',
};

const alignAttributes = ['alignX', 'alignY'];
const altitudeAttributes = ['altitude'];
const borderAttributes = ['border'];
const borderRadiusAttributes = ['borderRadius'];
const colorAttributes = [
  'color',
  'backgroundColor',
  'borderBlockEndColor',
  'borderBlockStartColor',
  'borderBottomColor',
  'borderColor',
  'borderInlineEndColor',
  'borderInlineStartColor',
  'borderLeftColor',
  'borderRightColor',
  'borderTopColor',
  'borderBottomColor',
  'caretColor',
  'columnRuleColor',
  'outlineColor',
  'textDecorationColor',
  'textEmphasisColor',
];
const spaceAttributes = [
  'margin',
  'marginLeft',
  'marginRight',
  'marginTop',
  'marginBottom',
  'marginX',
  'marginY',
  'padding',
  'paddingLeft',
  'paddingRight',
  'paddingTop',
  'paddingBottom',
  'paddingX',
  'paddingY',
  'top',
  'left',
  'bottom',
  'right',
  'grid-gap',
  'grid-column-gap',
  'grid-row-gap',
  'height',
  'width',
];
const fontAttributes = ['font', 'fontFamily'];
const fontSizeAttributes = ['fontSize'];
const fontWeightAttributes = ['fontWeight'];
const lineHeightAttributes = ['lineHeight'];
const letterSpacingAttributes = ['letterSpacing'];

const attributeMaps = {
  font: ['fontFamily'],
  marginY: ['marginTop', 'marginBottom'],
  paddingY: ['paddingTop', 'paddingBottom'],
  marginX: ['marginLeft', 'marginRight'],
  paddingX: ['paddingLeft', 'paddingRight'],
};

function getAlignmentStyles({ attribute, value, props }) {
  const alignYAttribute = props.display && props.display.includes('flex') ? 'align-items' : 'justify-content';
  const alignXAttribute = props.display && props.display.includes('flex') ? 'justify-content' : 'align-items';
  return css`
      display: flex;

      ${
        !props.display &&
        css`
          flex-direction: column;
        `
      }

      ${
        attribute === 'alignY' &&
        css`
          ${alignYAttribute}: ${FLEX_VERTICAL_ALIGN_MAP[value]};
        `
      }

      ${
        attribute === 'alignX' &&
        css`
          ${alignXAttribute}: ${FLEX_HORIZONTAL_ALIGN_MAP[value]};
        `
      }
    `;
}

function getAltitudeStyles({ theme, value }) {
  const altitudeStyle = altitude(value)({ theme });
  if (altitudeStyle) {
    return altitudeStyle;
  }
  return value;
}

function getBorderValue({ theme, value }) {
  const borderValue = border(value)({ theme });
  if (borderValue) {
    const borderColor = palette(borderValue.color)({ theme });
    return `${borderValue.width} ${borderValue.style} ${borderColor}`;
  }
  return value;
}

function getBorderRadiusValue({ theme, value }) {
  const borderRadiusValue = borderRadius(value)({ theme });
  if (borderRadiusValue) {
    return borderRadiusValue;
  }
  return value;
}

function getColorValue({ colorMode, theme, value }) {
  const color = palette(value)({ colorMode, theme });
  if (color) {
    return color;
  }
  return value;
}

function getSpaceValue({ theme, value }) {
  const spacing = space(value)({ theme });
  if (spacing) {
    return `${spacing}px`;
  }
  return value;
}

function getFontValue({ theme, value }) {
  const fontValue = font(value)({ theme });
  if (fontValue) {
    return fontValue;
  }
  return value;
}

function getFontSizeValue({ theme, value }) {
  const size = fontSize(value)({ theme });
  if (size) {
    return `${size}px`;
  }
  return value;
}

function getFontWeightValue({ theme, value }) {
  const weight = fontWeight(value)({ theme });
  if (weight) {
    return weight;
  }
  return value;
}

function getLineHeightValue({ theme, value }) {
  const height = lineHeight(value)({ theme });
  if (height) {
    return height;
  }
  return value;
}

function getLetterSpacingValue({ theme, value }) {
  const spacing = letterSpacing(value)({ theme });
  if (spacing) {
    return spacing;
  }
  return value;
}

export function getCSSFromStyleObject(props, theme, colorMode, { fromProps = false, disableCSSProps = [] } = {}) {
  let style = { ...props };
  if (style) {
    let styleEntries = Object.entries(style);
    styleEntries = styleEntries.reduce((prevStyle, [attribute, value]) => {
      let entries = [[attribute, value]];
      if (attributeMaps[attribute]) {
        entries = attributeMaps[attribute].map((attribute) => [attribute, value]);
      }
      return [...prevStyle, ...entries];
    }, []);

    style = styleEntries.reduce((prevStyle, [attribute, value]) => {
      let newValue = value;
      if (typeof value === 'function') {
        newValue = value({ theme });
      }
      if (attribute.includes(':')) {
        return css`
          ${prevStyle};

          ${attribute} {
            ${getCSSFromStyleObject(value, theme, colorMode, { fromProps })};
          }
        `;
      }
      if (typeof newValue === 'string') {
        newValue = { default: newValue };
      }
      if (attribute.includes('_')) {
        const platform = attribute.slice(1);
        return css`
          ${prevStyle};
          ${(Platform.OS === platform || (platform === 'native' && Platform.OS !== 'web')) &&
          getCSSFromStyleObject(value, theme, colorMode, { fromProps })}
        `;
      }
      const newStyle = Object.entries(newValue || {}).reduce((prevStyle, [platform, value]) => {
        let newStyle;
        let newValue = value;
        if (alignAttributes.includes(attribute)) {
          newStyle = getAlignmentStyles({ attribute, value, props });
          newValue = undefined;
        }
        if (altitudeAttributes.includes(attribute)) {
          newStyle = getAltitudeStyles({ theme, value });
          newValue = undefined;
        }
        if (borderAttributes.includes(attribute)) {
          newValue = getBorderValue({ theme, value });
        }
        if (borderRadiusAttributes.includes(attribute)) {
          newValue = getBorderRadiusValue({ theme, value });
        }
        if (colorAttributes.includes(attribute)) {
          newValue = getColorValue({ colorMode, theme, value });
        }
        if (spaceAttributes.includes(attribute)) {
          newValue = getSpaceValue({ theme, value });
        }
        if (fontAttributes.includes(attribute)) {
          newValue = getFontValue({ theme, value });
        }
        if (fontSizeAttributes.includes(attribute)) {
          if (disableCSSProps.includes('fontSize')) {
            return {};
          }
          newValue = getFontSizeValue({ theme, value });
        }
        if (fontWeightAttributes.includes(attribute)) {
          newValue = getFontWeightValue({ theme, value });
        }
        if (lineHeightAttributes.includes(attribute)) {
          newValue = getLineHeightValue({ theme, value });
        }
        if (letterSpacingAttributes.includes(attribute)) {
          newValue = getLetterSpacingValue({ theme, value });
        }
        if (platform === 'default' || Platform.OS === platform || (platform === 'native' && Platform.OS !== 'web')) {
          // @ts-ignore
          return css`
            ${prevStyle};
            ${newStyle};
            ${newValue
              ? css`
                  ${_kebabCase(attribute)}: ${newValue};
                `
              : ''}
          `;
        }
        return css`
          ${prevStyle};
          ${newStyle};
        `;
      }, css``);
      return css`
        ${prevStyle} ${newStyle};
      `;
    }, css``);
  }

  return style;
}
