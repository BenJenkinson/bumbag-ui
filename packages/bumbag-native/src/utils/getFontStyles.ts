import { font, fontWeight as getFontWeight } from './theme';

export function getFontAttributes({ fontFamilyThemeKey, fontWeight: _fontWeight, theme }) {
  let fontWeight = getFontWeight({ fontFamily: fontFamilyThemeKey, fontWeight: _fontWeight })({ theme });
  let fontFamily = font(fontFamilyThemeKey)({ theme });

  if (typeof fontWeight === 'object') {
    fontFamily = `${fontFamily}${fontWeight.fontFamilySuffix ? `-${fontWeight.fontFamilySuffix}` : ''}`;
    fontWeight = fontWeight.fontWeight || 'normal';
  }

  return {
    fontFamily,
    fontWeight,
  };
}

export function getFontStyles({ fontWeight: _fontWeight, fontFamily: _fontFamilyThemeKey = 'default' }) {
  return ({ theme, ...props }) => {
    const fontFamilyThemeKey = props.font || props.fontFamily || _fontFamilyThemeKey;

    const { fontFamily, fontWeight } = getFontAttributes({ fontFamilyThemeKey, fontWeight: _fontWeight, theme });

    return `
      ${fontFamily ? `font-family: ${fontFamily};` : ''}
      font-weight: ${fontWeight};
    `;
  };
}
