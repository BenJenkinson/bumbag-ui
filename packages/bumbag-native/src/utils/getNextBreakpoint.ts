import { theme as getTheme } from './theme';

export function getNextBreakpoint(breakpoint, { direction, theme }) {
  let breakpointSets = getTheme('breakpoints')({ theme })[direction];
  if (!Array.isArray(breakpointSets)) {
    breakpointSets = [breakpointSets];
  }

  let aboveBreakpoint;
  breakpointSets.forEach((breakpoints) => {
    const breakpointNames = Object.keys(breakpoints);
    breakpointNames.forEach((bp, i) => {
      if (bp === breakpoint) {
        aboveBreakpoint = breakpointNames[i + 1];
      }
    });
  });

  return aboveBreakpoint;
}
