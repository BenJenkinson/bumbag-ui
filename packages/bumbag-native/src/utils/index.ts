export * from './breakpoint';
export * from './getCSSFromStyleObject';
export * from './theme';
export * from './throttle';
export * from './useBreakpoint';
export * from './useStyle';
export * from './useTheme';
