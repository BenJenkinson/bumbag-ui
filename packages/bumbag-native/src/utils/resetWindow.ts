import { Platform } from 'react-native';

if (Platform.OS !== 'web') {
  window.addEventListener = () => {};
  window.removeEventListener = () => {};
}
