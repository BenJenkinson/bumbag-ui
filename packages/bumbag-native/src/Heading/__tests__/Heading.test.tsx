import * as React from 'react';
import { applyTheme } from 'bumbag/utils/applyTheme';
import { renderHook } from '@testing-library/react-hooks';
import { Heading } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Heading>Hello world</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Heading color="primary">Hello world</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H1', () => {
    const { container } = render(<Heading.H1>Hello world</Heading.H1>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H2', () => {
    const { container } = render(<Heading.H2>Hello world</Heading.H2>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H3', () => {
    const { container } = render(<Heading.H3>Hello world</Heading.H3>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H4', () => {
    const { container } = render(<Heading.H4>Hello world</Heading.H4>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H5', () => {
    const { container } = render(<Heading.H5>Hello world</Heading.H5>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H6', () => {
    const { container } = render(<Heading.H6>Hello world</Heading.H6>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('hook', () => {
    it('should return with Heading props', () => {
      const { result } = renderHook(() => Heading.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Heading>{(HeadingProps) => <Heading {...HeadingProps}>Hello world</Heading>}</Heading>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Heading.styles.base should render correctly', () => {
    const { container } = render(<Heading>hello world</Heading>, {
      theme: { Heading: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Heading.styles.base should render correctly', () => {
    const { container } = render(<Heading>Hello world</Heading>, {
      theme: {
        Heading: { styles: { base: (props) => ({ backgroundColor: props.color }) } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Heading.styles.base should render correctly for default font', () => {
    const { container } = render(<Heading font="default">hello world</Heading>, {
      theme: {
        fonts: { default: 'default', heading: 'heading' },
        Heading: { styles: { base: { backgroundColor: 'red' } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Heading.styles.base should render correctly for heading font', () => {
    const { container } = render(<Heading>hello world</Heading>, {
      theme: {
        fonts: { default: 'default', heading: 'heading' },
        Heading: { styles: { base: { backgroundColor: 'red' } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Heading.H1.styles.base should render correctly for H1', () => {
    const { container } = render(<Heading.H1>hello world</Heading.H1>, {
      theme: { Heading: { H1: { styles: { base: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Heading.H1.styles.base should render correctly for H1', () => {
    const { container } = render(<Heading.H1>Hello world</Heading.H1>, {
      theme: {
        Heading: { H1: { styles: { base: (props) => ({ backgroundColor: props.color }) } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Heading
        overrides={{
          Heading: {
            styles: { base: { backgroundColor: 'red' } },
            defaultProps: {
              altitude: '100',
              padding: 'major-2',
            },
          },
        }}
      >
        Hello world
      </Heading>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for H1', () => {
    const { container } = render(
      <Heading.H1
        overrides={{
          Heading: {
            H1: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: {
                altitude: '100',
                padding: 'major-2',
              },
            },
          },
        }}
      >
        Hello world
      </Heading.H1>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('applyTheme', () => {
  it('renders correctly', () => {
    const Card = applyTheme(Heading, {
      styles: {
        base: {
          backgroundColor: 'red',
        },
      },
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for H1', () => {
    const Card = applyTheme(Heading.H1, {
      styles: {
        base: {
          backgroundColor: 'red',
        },
      },
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<Heading variant="test">Hello world</Heading>, {
      theme: {
        Heading: {
          variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('styles.base should render correctly for H1', () => {
    const { container } = render(<Heading.H1 variant="test">Hello world</Heading.H1>, {
      theme: {
        Heading: {
          H1: {
            variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Heading
        overrides={{
          Heading: {
            variants: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        variant="test"
      >
        Hello world
      </Heading>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides for H1', () => {
    const { container } = render(
      <Heading.H1
        overrides={{
          Heading: {
            H1: {
              variants: {
                test: {
                  styles: { base: { backgroundColor: 'red' } },
                  defaultProps: {
                    altitude: '100',
                    padding: 'major-2',
                  },
                },
              },
            },
          },
        }}
        variant="test"
      >
        Hello world
      </Heading.H1>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Heading, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      variants: {
        test: {
          defaultProps: {
            color: 'red',
          },
        },
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme for H1', () => {
    const Card = applyTheme(Heading.H1, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      variants: {
        test: {
          defaultProps: {
            color: 'red',
          },
        },
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('modes', () => {
  it('should render correctly when colorMode is set globally', () => {
    const { container } = render(<Heading>Hello world</Heading>, {
      colorMode: 'test',
      theme: {
        Heading: {
          modes: {
            test: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: { color: 'primaryTint' },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when colorMode is set globally for H1', () => {
    const { container } = render(<Heading.H1>Hello world</Heading.H1>, {
      colorMode: 'test',
      theme: {
        Heading: {
          H1: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: { color: 'primaryTint' },
              },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when colorMode set as prop', () => {
    const { container } = render(<Heading colorMode="test">Hello world</Heading>, {
      theme: {
        Heading: {
          modes: {
            test: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: { color: 'primaryTint' },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when colorMode set as prop for H1', () => {
    const { container } = render(<Heading.H1 colorMode="test">Hello world</Heading.H1>, {
      theme: {
        Heading: {
          H1: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: { color: 'primaryTint' },
              },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Heading
        overrides={{
          Heading: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        colorMode="test"
      >
        Hello world
      </Heading>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides for H1', () => {
    const { container } = render(
      <Heading.H1
        overrides={{
          Heading: {
            H1: {
              modes: {
                test: {
                  styles: { base: { backgroundColor: 'red' } },
                  defaultProps: {
                    altitude: '100',
                    padding: 'major-2',
                  },
                },
              },
            },
          },
        }}
        colorMode="test"
      >
        Hello world
      </Heading.H1>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Heading, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      modes: {
        test: {
          styles: { base: { backgroundColor: 'red' } },
          defaultProps: { color: 'primaryTint' },
        },
      },
    });
    const { container } = render(<Card colorMode="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme for H1', () => {
    const Card = applyTheme(Heading.H1, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      modes: {
        test: {
          styles: { base: { backgroundColor: 'red' } },
          defaultProps: { color: 'primaryTint' },
        },
      },
    });
    const { container } = render(<Card colorMode="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for color', () => {
    const { container } = render(<Heading>hello world</Heading>, {
      theme: { Heading: { defaultProps: { color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for color for H1', () => {
    const { container } = render(<Heading>hello world</Heading>, {
      theme: { Heading: { H1: { defaultProps: { color: 'primary' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
