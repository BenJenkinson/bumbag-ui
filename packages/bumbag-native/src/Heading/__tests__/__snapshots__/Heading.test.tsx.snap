// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyTheme renders correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 24px; font-weight: 700; padding: 16px 16px 16px 16px;"
/>
`;

exports[`applyTheme renders correctly for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 48px; font-weight: 700; padding: 16px 16px 16px 16px;"
/>
`;

exports[`composition hook should return with Heading props 1`] = `
Object {
  "active": false,
  "colorMode": "default",
  "elementRef": [Function],
  "focus": false,
  "hover": false,
  "hoveractive": false,
  "ref": [Function],
  "style": Array [
    Object {},
  ],
}
`;

exports[`composition render props should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for color 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(87, 79, 235); font-size: 24px; font-weight: 700;"
>
  hello world
</div>
`;

exports[`defaultProps should render correctly for color for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  hello world
</div>
`;

exports[`modes renders correctly for applyTheme 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(238, 237, 253); font-size: 24px; font-weight: 700; padding: 16px 16px 16px 16px;"
/>
`;

exports[`modes renders correctly for applyTheme for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(238, 237, 253); font-size: 48px; font-weight: 700; padding: 16px 16px 16px 16px;"
/>
`;

exports[`modes renders correctly for overrides 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 24px; font-weight: 700; padding: 16px 16px 16px 16px;"
>
  Hello world
</div>
`;

exports[`modes renders correctly for overrides for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 48px; font-weight: 700; padding: 16px 16px 16px 16px;"
>
  Hello world
</div>
`;

exports[`modes should render correctly when colorMode is set globally 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(238, 237, 253); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`modes should render correctly when colorMode is set globally for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(238, 237, 253); font-size: 48px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`modes should render correctly when colorMode set as prop 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(238, 237, 253); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`modes should render correctly when colorMode set as prop for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(238, 237, 253); font-size: 48px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`overrides should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 24px; font-weight: 700; padding: 16px 16px 16px 16px;"
>
  Hello world
</div>
`;

exports[`overrides should render correctly for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 48px; font-weight: 700; padding: 16px 16px 16px 16px;"
>
  Hello world
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 48px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly for H2 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 40px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly for H3 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 32px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly for H4 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly for H5 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 20px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly for H6 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 16px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(87, 79, 235); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`theming Heading.H1.styles.base should render correctly for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(33, 33, 33); font-size: 48px; font-weight: 700;"
>
  hello world
</div>
`;

exports[`theming Heading.H1.styles.base should render correctly for H1 2`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 48px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`theming Heading.styles.base should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  hello world
</div>
`;

exports[`theming Heading.styles.base should render correctly 2`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`theming Heading.styles.base should render correctly for default font 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(33, 33, 33); font-family: default; font-size: 24px; font-weight: 700;"
>
  hello world
</div>
`;

exports[`theming Heading.styles.base should render correctly for heading font 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(33, 33, 33); font-family: heading; font-size: 24px; font-weight: 700;"
>
  hello world
</div>
`;

exports[`variants renders correctly for applyTheme 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(255, 0, 0); font-size: 24px; font-weight: 700; padding: 16px 16px 16px 16px;"
/>
`;

exports[`variants renders correctly for applyTheme for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(255, 0, 0); font-size: 48px; font-weight: 700; padding: 16px 16px 16px 16px;"
/>
`;

exports[`variants renders correctly for overrides 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 24px; font-weight: 700; padding: 16px 16px 16px 16px;"
>
  Hello world
</div>
`;

exports[`variants renders correctly for overrides for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(33, 33, 33); font-size: 48px; font-weight: 700; padding: 16px 16px 16px 16px;"
>
  Hello world
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(33, 33, 33); font-size: 24px; font-weight: 700;"
>
  Hello world
</div>
`;

exports[`variants styles.base should render correctly for H1 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); color: rgb(33, 33, 33); font-size: 48px; font-weight: 700;"
>
  Hello world
</div>
`;
